pkg/processor/runtime/rpc/abstract.go

```diff
@@ -301,34 +244,20 @@ func (r *AbstractRuntime) Terminate() error {
 	return nil
 }
 
-func (r *AbstractRuntime) processItemAndWaitForResult(item interface{}, functionLogger logger.Logger) (*batchedResults, error) {
+func (r *AbstractRuntime) allocateSocketAndWaitForResult(item interface{}, functionLogger logger.Logger) (*batchedResults, error) {
 
 	if currentStatus := r.GetStatus(); currentStatus != status.Ready {
 		return nil, errors.Errorf("Processor not ready (current status: %s)", currentStatus)
 	}
 
-	r.functionLogger = functionLogger
+	socket := r.socketAllocator.Allocate()
 
-	// We don't use defer to reset r.functionLogger since it decreases performance
-	if err := r.eventEncoder.Encode(item); err != nil {
+	r.functionLogger = functionLogger
```

Comment:
does the `defer` not affect the performance anymore?

Reply:
?

Reply:
@TomerShor Comment is below